File: numbers/prime_factors/prime_factors.py
import math

def PrimeFactors(num):
	primeFactors = []
	for i in range(2, int(math.sqrt(num)) + 1):
		while num % i == 0:
			primeFactors.append(i)
			num //= i
	if num > 2:
		primeFactors.append(num)
	return primeFactors

def main():
	factors = PrimeFactors(36)
	print(factors)

if __name__ == '__main__':
	main()

File: numbers/prime_factors/prime_factors.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

vector<int> prime_factors(int num) {

	vector<int> factors;
	for (int i = 2; i <= sqrt(num); i++) {
		while (num % i == 0) {
			factors.push_back(i);
			num /= i;
		}
	}
	if (num > 2)
		factors.push_back(num);
	return factors;
}

int main() {

	vector<int> factors = prime_factors(156);
	for (int i = 0; i < factors.size(); i++)
		cout << factors[i] << " ";
	return 0;
}
